rdtools/system_analysis.py

```diff
@@ -0,0 +1,587 @@
+'''
+This module contains functions and classes for object-oriented end-to-end analysis
+'''
+import pvlib
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from . import normalization
+from . import filtering
+from . import aggregation
+from . import degradation
+from . import soiling
+from . import clearsky_temperature
+from . import plotting
+
+
+class system_analysis():
+    '''
+    Class for end-to-end analysis
+
+    Instantiation parameters
+    ----------
+    pv: Right-labeled Pandas Time Series of PV energy or power. If energy, should *not* be cumulative, but only for preceding time step.
+    poa: Right-labeled Pandas Time Series of measured plane of array irradiance in W/m^2
+    temperature: Right-label Pandas Time Series of either cell or ambient temperature in Celsius
+    temperature_coefficient: The fractional PV power temperature coefficient (numeric)
+    aggregation_freq: The frequency with which to aggregate normalized PV data for analysis (Pandas frequency specification)
+    pv_input: 'power' or 'energy' to specify type of input used for pv parameter
+    temeprature_input: 'cell' or 'ambient' to specify type of input used for temperature parameter
```

Comment:
We should consider changing the options for temperature input to `module` and `ambient`. In practice it is a good approximation to use back of module temperature for temperature corrections. If we do make this change, we should also change the behavior and name of `system_analysis.calc_cell_temperature()` to return module temperature.

Reply:
I agree, both option would be permissible. Module temperature is more accurate but for degradation analysis consistency is important. In practice, I have seen a lot of back-of-the-module sensors peel off over many years in the field. This can bias the results. 
On a different note, the description of the temperature coefficient could be a little bit more specific. For example, if the temp coefficient is 0.40 %/year is the numeric input to be used 0.4 or 0.004, probably 0.004 but to a beginner this may not be clear.

Reply:
Since the functional API still used cell temperature, I suggest we put a pin in this (open an issue) and proceed with cell temperature for now.